### PR TITLE
Main Project Cleanup

### PR DESCRIPTION
**Pull request by paulushub**
### Description
- Clean up of the main source code project
- Removed irrelevant files
- Added support for .NET 4.7, .NET 4.8
- Removed preprocessor to enable events registration for all
- Updates to the `SvgOptions` class to support key-value properties. Will use it to resolve issue raised in #987 
- Split `ItemGroup` for references into supported Frameworks, so dependent package versions can be separately updated.
- Resolved main concern in #1019 

### Type of change
- [x] Removed https://svg.codeplex.com/ and https://svg.codeplex.com/license from description.
    Why? Links no longer exist.
- [x] Removed Microsoft.NETFramework.ReferenceAssemblies
    Why? SDK-style projects include this reference by default. 
    Ref: https://learn.microsoft.com/en-us/dotnet/framework/migration-guide/reference-assemblies
- [x] Dropped `<Title>Svg for .Net Framework 4.6.2</Title>` in project file for simplicity.
    - Why? Per-package title is not used/displayed by Visual Studio or nuget.org
- [x] Removed `DisableImplicitFrameworkReferences` for .NET 4.6.2
    - Why? Not applicable
- [x] Removed .NET 4 support preprocessors, like
  - `Enum.TryParse`: Is supported on all current platforms
  - string.Join(String, IEnumerable<String>):  Is supported on all current platforms
- [x] Dropped Svg.Enums.TryParse for System.Enum.TryParse: No longer needed
- [x] Removed irrelevant files
   - **NuGet.Config**: Empty NuGet configuration file
   - **Svg.csproj.vspscc**: Old MS Teams file
   - **Svg.nuspec**: Unsed NuGet package file
   - **Svg.sln.DotSettings**: Empty reshaper file
   - **Svg.vsmdi**: MSTest file
   - **Local.testsettings:** MSTest file
   - **TraceAndTestImpact.testsettings**: MSTest file
- [x] Clean up (non-breaking change which fixes an issue)
- [x] This change does not require a documentation update
- [x] Resolved review issues.

### How Has This Been Tested?
- [x] SvgW3CTestRunner is tested for each supported platform
- [x] Svg.UnitTests tested on command line for each supported platform
- [x] Svg.Benchmarks tested on command line for each supported platform, warnings are as before will be addressed later.

### CODE REVIEW

File: Samples/Entities/Entities.csproj [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+		<NoWarn>$(NoWarn);CS0618;NETSDK1138</NoWarn>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+		<NoWarn>$(NoWarn);CS1591;CS0618;NETSDK1138</NoWarn>
+	</PropertyGroup>
+
+	<ItemGroup>
+		<ProjectReference Include="..\..\Source\Svg.csproj" />
+	</ItemGroup>
```
Comment (mrbean-bremen) [inline Samples/Entities/Entities.csproj:17]: You have changed spaces to tabs in the project files. Is this intentional?

File: Source/DataTypes/EnumConverters.cs [hunk 2]
```diff
@@ -220,21 +220,21 @@ public sealed class SvgStitchTypeConverter : EnumBaseConverter<SvgStitchType> {
...
+    //            result = (TEnum)Enum.Parse(typeof(TEnum), value, true);
+    //            return true;
+    //        }
+    //        catch
+    //        {
+    //            result = default(TEnum);
+    //            return false;
+    //        }
+    //    }
+    //}
```
Comment (mrbean-bremen) [inline Source/DataTypes/EnumConverters.cs:239]: Is this not needed bevause you just use `Enum`? In this case just remove it. 
@H1Gdev may have a look as he introduced this.

File: Source/Svg.csproj [hunk 3]
```diff
@@ -1,102 +1,151 @@
...
+        <Optimize>true</Optimize>
+        <DefineConstants>TRACE</DefineConstants>
+        <NoWarn>$(NoWarn);CS1591;CS0618;NETSDK1138</NoWarn>
+        <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    </PropertyGroup>
+    
+    <PropertyGroup>
+        <DefineConstants Condition="$(TargetFramework.StartsWith('net46'))">$(DefineConstants);DOTNET46;NETFULL</DefineConstants>
+        <DefineConstants Condition="$(TargetFramework.StartsWith('net47'))">$(DefineConstants);DOTNET47;NETFULL</DefineConstants>
+        <DefineConstants Condition="$(TargetFramework.StartsWith('net48'))">$(DefineConstants);DOTNET48;NETFULL</DefineConstants>
```
Comment (mrbean-bremen) [inline Source/Svg.csproj:61]: Why are these needed?

File: Tests/Svg.Benchmark/Svg.Benchmark.csproj [hunk 4]
```diff
@@ -1,45 +1,52 @@
...
-	<AssemblyOriginatorKeyFile>..\Svg.UnitTests\svgkey.snk</AssemblyOriginatorKeyFile>
-  </PropertyGroup>
+	<PropertyGroup>
+		<OutputType>Exe</OutputType>
+		<TargetFrameworks>net6.0;netcoreapp3.1;net462</TargetFrameworks>
+		<IsPackable>false</IsPackable>
+		<LangVersion>8.0</LangVersion>
+		<SignAssembly>True</SignAssembly>
+		<AssemblyOriginatorKeyFile>..\Svg.UnitTests\svgkey.snk</AssemblyOriginatorKeyFile>
+	</PropertyGroup>
```
Comment (mrbean-bremen) [inline Tests/Svg.Benchmark/Svg.Benchmark.csproj:9]: This one again has move to tabs.

File: Samples/Entities/Entities.csproj (hunk 1 above)
Comment (paulushub) [reply Samples/Entities/Entities.csproj:17]: Yes, using the default 4 spaces tab makes it easier to edit the files. The default indentation size in VS is 4 spaces unless changed by a user.
The project file indentation was 2 spaces.

File: Source/DataTypes/EnumConverters.cs (hunk 2 above)
Comment (paulushub) [reply Source/DataTypes/EnumConverters.cs:239]: Thanks, we wait for @H1Gdev. 
The generic support for the parsing `enum` string was not available in 3.5, so that might be needed but no more.

File: Tests/Svg.Benchmark/Svg.Benchmark.csproj (hunk 4 above)
Comment (paulushub) [reply Tests/Svg.Benchmark/Svg.Benchmark.csproj:9]: Tabs are not intended. 4 spaces is will fix that. It seems on my new developer PC, tap -> spaces is not enabled,

File: Source/Svg.csproj (hunk 3 above)
Comment (paulushub) [reply Source/Svg.csproj:61]: Added support for .NET Framework 4.7 and 4.8, for project using them.
The constants are provided in case any platform specific coding is necessary. Currently, only `NETFULL` is used.

**Comment by paulushub** [comment]
@mrbean-bremen Sorry for the crazy tabs, I have fixed it. 
I only applied `Edit -> Advanced -> Format Document`, but unfortunately the tabify option is still on in my current machine.

File: Source/Svg.csproj (hunk 3 above)
Comment (mrbean-bremen) [reply Source/Svg.csproj:61]: Ok, as long as there are no additional projects are build by default, it won't actually matter, so it is ok, I guess.

**Review by mrbean-bremen — COMMENTED** [review]
Looks good to me, but I would want to have @H1Gdev have another look.

**Comment by paulushub** [comment]
> Looks good to me, but I would want to have @H1Gdev have another look.

Thanks, trying to add him to the reviewers, but the name is not showing up.

**Comment by mrbean-bremen** [comment]
> Thanks, trying to add him to the reviewers, but the name is not showing up.

I also had tried to add him to the contributors. Anyway, he will get notifications from mentioning. And I think he is in a similar timezone as you, so it should be easier for you to communicate ;)

**Comment by paulushub** [comment]
@mrbean-bremen  Unfortunately, not all interpolation modes will allow the SVG-log to pass the tests. So, I set the default as `Default`, let any user who wants to experiment with his/her desire mode of interpolation to use the `SvgOptions`.
Since this is not a W3C test file, I do not know how the reference PNG is created.

image

File: Source/DataTypes/EnumConverters.cs (hunk 2 above)
Comment (H1Gdev) [reply Source/DataTypes/EnumConverters.cs:239]: @paulushub @mrbean-bremen 

I agree with remove it.
It would be beneficial to be able to remove `[CLSCompliant(false)]`.

Comment (paulushub) [reply Source/DataTypes/EnumConverters.cs:239]: @H1Gdev Thanks for the feedback. I will now remove the commented code.

File: Source/Rendering/SvgRenderer.cs [hunk 5]
```diff
@@ -135,7 +135,7 @@ private static Graphics CreateGraphics(Image image)
             g.CompositingQuality = CompositingQuality.HighQuality;
             g.TextRenderingHint = TextRenderingHint.AntiAlias;
             g.TextContrast = 1;
-            g.InterpolationMode = InterpolationMode.NearestNeighbor;
```
Comment (H1Gdev) [inline Source/Rendering/SvgRenderer.cs:138]: @paulushub 

Do you need to change this setting ?
I recommend that you change this setting carefully, such as by creating another PR.

File: Source/SvgOptions.cs [hunk 6]
```diff
@@ -1,31 +1,177 @@
...
 
-#nullable enable
-
-namespace Svg;
-
-public class SvgOptions
+namespace Svg
 {
-    public SvgOptions()
+    public class SvgOptions : IDictionary<string, string>, ICloneable
```
Comment (H1Gdev) [inline Source/SvgOptions.cs:10]: I think it would be better to inherit `Dictionary`.

File: Source/Rendering/SvgRenderer.cs (hunk 5 above)
Comment (paulushub) [reply Source/Rendering/SvgRenderer.cs:138]: It was added by a PR we allowed to record the contribution and close the issue as we planned to improve the implementation.
However, as you can see it is causing the build to fail with a SVG-Logo. 
So, I set it to `Default` to make all the tests pass as before.

File: Source/SvgOptions.cs (hunk 6 above)
Comment (paulushub) [reply Source/SvgOptions.cs:10]: Just wanted it to be as dictionary, we might change the base (and internal) to anything later.

**Comment by paulushub** [comment]
@mrbean-bremen @H1Gdev Sorry, if it OK with you, I will merge this PR.

**Review by mrbean-bremen — APPROVED** [review]

**Review by H1Gdev — APPROVED** [review]
